store/storetest/user_store.go

```diff
@@ -1056,9 +1056,12 @@ func testUserStoreGetAllProfilesInChannel(t *testing.T, ss store.Store) {
 		var profiles map[string]*model.User
 		profiles, err = ss.User().GetAllProfilesInChannel(c2.Id, true)
 		require.Nil(t, err)
-		assert.Equal(t, map[string]*model.User{
-			u1.Id: sanitized(u1),
-		}, profiles)
+		require.Equal(t, 1, len(profiles))
+		profile, ok := profiles[u1.Id]
+		require.True(t, ok)
+		require.True(t, profile.AuthData == nil || *profile.AuthData == "")
+		profile.AuthData = model.NewString("")
+		require.Equal(t, sanitized(u1), profile)
```

Comment:
This block of change is to workaround [an issue in gob deserialization](https://stackoverflow.com/questions/42675054/golang-gob-converts-pointer-to-0-into-nil-pointer) as it does not distinguish between `nil` and `""` for data type `*string` unfortunately. 

I have scanned through the code base and it looks like `AuthData == nil` and `AuthData == ""` represent the same meaning throughout the code base.

Reply:
Wondering if this is still needed?

Reply:
just tried we dont. i am reverting this back.